install/scripts/rebuild-base-images.sh

```diff
@@ -1,7 +1,25 @@
 #!/bin/bash
-echo "Building Ubuntu 18.04 image..."
-sudo singularity build -F base-ubuntu-18.04.sif base-ubuntu-18.04.def
+set -e
+HIPSYCL_PKG_SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+HIPSYCL_PKG_CONTAINER_DIR=${HIPSYCL_PKG_CONTAINER_DIR:-$HIPSYCL_PKG_SCRIPT_DIR/containers}
+HIPSYCL_PKG_LLVM_REPO_BRANCH=${HIPSYCL_PKG_LLVM_REPO_BRANCH:-release/9.x}
+mkdir -p /tmp/hipsycl-pkg-builder
+cp $HIPSYCL_PKG_SCRIPT_DIR/*.def /tmp/hipsycl-pkg-builder
+cp $HIPSYCL_PKG_SCRIPT_DIR/*.sh /tmp/hipsycl-pkg-builder
+cd /tmp/hipsycl-pkg-builder
+
+echo $HIPSYCL_PKG_CONTAINER_DIR
+
+# Workaround for exposing variable inside singularity post script
+sed -i "/%post/a export\ HIPSYCL_PKG_LLVM_REPO_BRANCH=$HIPSYCL_PKG_LLVM_REPO_BRANCH" base-ubuntu-18.04.def
+sed -i "/%post/a export\ HIPSYCL_PKG_LLVM_REPO_BRANCH=$HIPSYCL_PKG_LLVM_REPO_BRANCH" base-archlinux-rolling.def
+sed -i "/%post/a export\ HIPSYCL_PKG_LLVM_REPO_BRANCH=$HIPSYCL_PKG_LLVM_REPO_BRANCH" base-centos-7.def
```

Comment:
Does `SINGULARITYENV_XYZ` not work to have environment variable `$XYZ` in the post script? Or is this only available after build time?

Reply:
As far as I know, it will only be available after build time, unfortunately...

Reply:
I wonder if there's a clean solution.. This all sounds very hacky to me.

Reply:
I am sorry, you were right with the SINGULARITYENV_XYZ variable one can expose vars for the postscript as well